### DISCUSSION TITLE
Reduce the number of required fields

### DISCUSSION THREAD

**Discussion by lehors** [lfai/model_openness_tool#24] [General]
While I understand the interest in trying to get people to give as much info as possible, requiring a lot of information for a model to be entered in the system creates a barrier that is likely to deter people from using the system altogether.

As it stands none of the models we currently have in the system meets the requirements enforced by the UI. This is a clear sign that we are asking for too much.

The models currently in the system only are in it because they were imported through a function that doesn't enforce those same requirements. Any attempt to edit one of the existing models via the UI is met with a strict enforcement of these requirements which effectively makes it impossible to just add one piece of missing data. All missing information needs to be added for the update to be accepted.

I think we need to reduce the number of required fields to what is necessary for the system to function rather than to what we'd like people to enter.

**Comment by gregcube** [comment]
The purpose of the `Admin` tab (visible only to site admins) is to allow model edits without validation, while the `Edit` tab enforces field requirements for regular users. However, I just noticed that the admin bypass isn't working as expected (validation still runs).

I do see your point -- there is a lot of information required for regular users to fully submit a model, and I agree that this could discourage them from using the system. For the system to function we would only need a model name and organization.  Organization may not even be necessary(?) -- it would just be empty on the /models page if left blank.  An unique model name and component fields are the minimum requirements for evaluation to run and to save the model.

**Reply by lehors** [reply]
Yes, I actually played with that and I think it makes the system much more user friendly. See PR #25 

Even though it's not strictly required I opted to keep the org name in addition to the model name because the filter provided on the model page allows one to enter an organization name, so it is consistent.

**Comment by lehors** [comment]
So, I didn't realize the Admin model edit function was meant to behave differently. I'll need to think more through it but it doesn't seem desirable to have different behaviors in that way.

**Reply by gregcube** [reply]
After reviewing this again, I should also mention that using the `Admin` edit form, an admin can assign any type of license to any component; they're not restricted to code, data or document. This was originally implemented so admins can assign open source licenses to `Model parameters (Final)`.  Validation is bypassed for all fields except those under model details; however, it should bypass those fields as well (it's a bug).

**Comment by lehors** [comment]
This has been addressed and only the model identifier is actually required now.
